src/doc-printer.js

```diff
@@ -387,12 +392,13 @@ function printDocToString(doc, options) {
                   // Trim whitespace at the end of line
                   while (
                     out.length > 0 &&
+                    out[out.length - 1].match &&
```

Comment:
This is a quick/dirty way of getting the tests to pass, and should probably be replaced with more robust logic that skips over cursor placeholder Symbols.